doc/user-manual/language/syntactic-sugar.lagda.rst

```diff
@@ -61,6 +61,22 @@ Syntactic Sugar
    :depth: 2
    :local:
 
+.. _hidden_argument_puns:
+
+Hidden argument puns
+====================
+
+If the option :option:`--hidden-argument-puns` is used, then the
+pattern ``{x}`` is interpreted as ``{x = x}``, and the pattern
```

Comment:
I suppose "pattern" is supposed to mean "pattern in the lhs of a function definition". Please make this clear here.  We have patterns also in Pis and lets, or even in module telescopes.

Reply:
I wrote that "I would not be surprised if it turns out that I missed something", and this is one such thing. I think that puns should be supported in Π-types, let expressions and module telescopes, and I'll add support for that. Is there anything else that I've missed?

Reply:
It should also be possible to use puns in do notation, in rewrite equations, in pattern synonyms, and perhaps in display pragmas.

It also makes sense to support puns in sub-patterns in lambda-bindings. However, I don't currently plan to implement support for puns in top-level patterns in lambda abstractions or the left-hand sides of syntax declarations. These constructions (which use `LamBinding`) currently allow the use of `{x = y}`, but the name `x` seems to be ignored. For instance, the following code is rejected:
```agda
_ : (A : Set) → {B : Set} {x : A} → A
_ = λ _ {x = x} → x
```

I'll update the documentation.

Reply:
> It should also be possible to use puns in do notation, in rewrite equations, in pattern synonyms, and perhaps in display pragmas.
>
> It also makes sense to support puns in sub-patterns in lambda-bindings.

I've implemented something like this (also for display pragmas) and added test cases.